utils/hct/hctdb.py

```diff
@@ -6110,31 +6160,52 @@ def finalize_dxil_operations(self):
                 )
                 self.verify_dense(i.ops, lambda x: x.pos, lambda x: i.name)
 
-        # Verify that all operations in each class have the same signature.
+        self.verify_dxil_op_classes()
+
+    def verify_dxil_op_classes(self):
+        "Verify that all DXIL operations in each class have the same signature."
         import itertools
 
-        class_sort_func = lambda x, y: x < y
         class_key_func = lambda x: x.dxil_class
         instr_ordered_by_class = sorted(self.get_dxil_ops(), key=class_key_func)
         instr_grouped_by_class = itertools.groupby(
             instr_ordered_by_class, key=class_key_func
         )
 
         def calc_oload_sig(inst):
-            result = ""
-            for o in inst.ops:
-                result += o.llvm_type
+            # if function class is ever overloaded, no "v" oload should be used
+            oload = inst.oload_types == "v" and "void" or "overloaded"
```

Comment:
nit: I think a ternary operator would be clearer, but didn't know python allowed this.